src/components/StakingVaults/WalletProviderTable.tsx

```diff
@@ -56,22 +56,19 @@ export const WalletProviderTable: React.FC<ProviderTableProps> = ({
     }
     return (
       <Flex gap={4} flexDir='column'>
+        {
+          // Assume three max loading rows - that might not be true, but we don't want to collapse the
+          // loaded rows too much and hinder visibility
+          Array.from({ length: Math.min(pendingOpportunityApis, 3) }).map((_, index) => (
```

Comment:
To be confirmed by big @reallybeard - alternatively, we could make this a static number, revert back the selector to a boolean and call it a day making this the only diff, which would make it less collapsy as requests go through